app/controlplane/internal/usercontext/attjwtmiddleware/attmiddleware.go

```diff
@@ -0,0 +1,269 @@
+	ErrTokenParseFail         = errorsAPI.Unauthorized(reason, "Fail to parse JWT token ")
+	ErrUnSupportSigningMethod = errorsAPI.Unauthorized(reason, "Wrong signing method")
+	ErrWrongContext           = errorsAPI.Unauthorized(reason, "Wrong context for middleware")
+)
+
+// NewAttestationRobotAccountProvider return the configuration to validate and verify token issued for Robot Accounts
+func NewAttestationRobotAccountProvider(signingSecret string) JWTOption {
+	return WithTokenProvider(
+		RobotAccountProviderKey,
+		WithClaims(func() jwt.Claims { return &robotaccount.CustomClaims{} }),
+		WithVerifyAudienceFunc(func(token *jwt.Token) bool {
+			claims, ok := token.Claims.(*robotaccount.CustomClaims)
+			if !ok {
+				return false
+			}
+			for _, aud := range []string{robotaccount.Audience, robotaccount.DeprecatedAudience} {
+				if claims.VerifyAudience(aud, true) {
+					return true
+				}
+			}
+			return false
+		}),
+		WithSigningMethod(robotaccount.SigningMethod),
+		WithKeyFunc(func(_ *jwt.Token) (interface{}, error) {
+			// TODO: add support to multiple signing methods and keys
+			return []byte(signingSecret), nil
+		}),
+	)
+}
+
+// NewAttestationAPITokenProvider return the configuration to validate and verify token issued for API Tokens
+func NewAttestationAPITokenProvider(signingSecret string) JWTOption {
+	return WithTokenProvider(
+		APITokenProviderKey,
+		WithVerifyAudienceFunc(func(token *jwt.Token) bool {
+			claims, ok := token.Claims.(jwt.MapClaims)
+			if !ok {
+				return false
+			}
+			for _, aud := range []string{apitoken.Audience} {
+				if claims.VerifyAudience(aud, true) {
+					return true
+				}
+			}
+			return false
+		}),
+		WithSigningMethod(user.SigningMethod),
+		WithKeyFunc(func(_ *jwt.Token) (interface{}, error) {
+			return []byte(signingSecret), nil
+		}),
+	)
+}
+
+type JWTAuthContext struct {
+	Claims      jwt.Claims
+	ProviderKey string
+}
+
+type authzContextKey struct{}
+
+type JWTOption func(*options)
+type TokenProviderOption func(*providerOption)
+
+type VerifyAudienceFunc func(*jwt.Token) bool
+
+type providerOption struct {
+	providerKey        string
+	signingMethod      jwt.SigningMethod
+	keyFunc            jwt.Keyfunc
+	claims             func() jwt.Claims
+	verifyAudienceFunc VerifyAudienceFunc
+}
+
+func WithSigningMethod(method jwt.SigningMethod) TokenProviderOption {
+	return func(o *providerOption) {
+		o.signingMethod = method
+	}
+}
+
+func WithClaims(f func() jwt.Claims) TokenProviderOption {
+	return func(o *providerOption) {
+		o.claims = f
+	}
+}
+
+func WithKeyFunc(keyFunc jwt.Keyfunc) TokenProviderOption {
+	return func(o *providerOption) {
+		o.keyFunc = keyFunc
+	}
+}
+
+func WithVerifyAudienceFunc(f VerifyAudienceFunc) TokenProviderOption {
+	return func(o *providerOption) {
+		o.verifyAudienceFunc = f
+	}
+}
+
+type options struct {
+	tokenProviders []providerOption
+}
+
+func WithTokenProvider(providerKey string, opts ...TokenProviderOption) JWTOption {
```

Comment:
WithTokenProvider doens't need to be exposed anymore, let's make it downcase no?